src/clojure/nrepl/transport.clj

```diff
@@ -252,3 +252,27 @@
   "Send a response for `msg` with `response-data` using message's transport."
   [msg & response-data]
   (send (:transport msg) (apply response-for msg response-data)))
+
+(defmacro safe-handle
```

Comment:
Perhaps `with-safe-handlers` would be slightly better name for this macro, but I'm not totally certain about this.

Reply:
I don't like `with-safe-handlers` here (or any `with-` for that matter) because it not only establishes some context, like `with-` macros do, but actually performs an action – invokes the appropriate handler.